src/anyio/pytest_plugin.py

```diff
@@ -70,28 +71,56 @@ def pytest_configure(config: Any) -> None:
     )
 
 
-def pytest_fixture_setup(fixturedef: Any, request: Any) -> None:
-    def wrapper(*args, anyio_backend, **kwargs):  # type: ignore[no-untyped-def]
+@pytest.hookimpl(hookwrapper=True)
+def pytest_fixture_setup(fixturedef: Any, request: Any) -> Generator[Any]:
+    def wrapper(
+        *args: Any, anyio_backend: Any, request: SubRequest, **kwargs: Any
+    ) -> Any:
+        # Rebind any fixture methods to the request instance
+        if (
+            request.instance
+            and ismethod(func)
+            and type(func.__self__) is type(request.instance)
+        ):
+            local_func = func.__func__.__get__(request.instance)
+        else:
+            local_func = func
+
         backend_name, backend_options = extract_backend_and_options(anyio_backend)
         if has_backend_arg:
             kwargs["anyio_backend"] = anyio_backend
 
+        if has_request_arg:
+            kwargs["request"] = anyio_backend
```

Comment:
This overwrites the `request` param of all async fixtures so that you are no longer able to parameterise async fixtures at all. The following example with throw an `AttributeError: 'str' object has no attribute 'param'`:

```py
@pytest.fixture(params=['foo', 'bar'])
async def dummy_fixture(
    request: pytest.FixtureRequest,
):
    if request.param == 'foo':
        do_this()
    else:
        do_that()
```

Reply:
Seems like this was missed in the review. I'll make a fix promptly.

Reply:
I've released a fix now.

Reply:
Thank you!